src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1363,6 +1351,8 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
    std::string encoding = projects::projectContext().hasProject() ?
                           projects::projectContext().defaultEncoding() :
                           prefs::userPrefs().defaultEncoding();
+   if (encoding.empty())
+      encoding = "UTF-8";
    std::string encodedString;
    error = r::util::iconvstr(grepOptions.searchPattern(),
```

Comment:
Do we need to run `iconvstr` if the target encoding is UTF-8? (seems like asking it to convert UTF-8 to UTF-8 would be a noop)

Reply:
It's not a noop but possibly overkill, it sets `encodedString = grepOptions.searchPattern()` and will return early before attempting any conversion logic. `encodedString` writes the string in a temp file to be referenced later on.